src/token.ts

```diff
@@ -194,34 +172,93 @@ export const isTooEarly = (ucan: Ucan): boolean => {
   return ucan.payload.nbf > Math.floor(Date.now() / 1000)
 }
 
+
+export interface ValidateOptions {
+  checkSignature?: boolean
+  checkIsExpired?: boolean
+  checkIsTooEarly?: boolean
+}
+
 /**
- * Check if a UCAN is valid.
- *
- * @param ucan The decoded UCAN
- * @param did The DID associated with the signature of the UCAN
+ * Parse & Validate **one layer** of a UCAN.
+ * This doesn't validate attenutations and doesn't validate the whole UCAN chain.
+ * 
+ * By default, this will check the signature and time bounds.
+ * 
+ * @param encodedUcan the JWT-encoded UCAN to validate
+ * @param options an optional parameter to configure turning off some validation options
+ * @returns the parsed & validated UCAN (one layer)
+ * @throws Error if the UCAN is invalid
  */
- export async function isValid(ucan: Ucan): Promise<boolean> {
-  const encodedHeader = encodeHeader(ucan.header)
-  const encodedPayload = encodePayload(ucan.payload)
+export async function validate(encodedUcan: string, options?: ValidateOptions): Promise<Ucan> {
+  const checkSignature = options?.checkSignature ?? true
+  const checkIsExpired = options?.checkIsExpired ?? true
+  const checkIsTooEarly = options?.checkIsTooEarly ?? true
+
+  const [encodedHeader, encodedPayload, signature] = encodedUcan.split(".")
+  if (encodedHeader == null || encodedPayload == null || signature == null) {
+    throw new Error(`Can't parse UCAN: ${encodedUcan}: Expected JWT format: 3 dot-separated base64url-encoded values.`)
+  }
 
-  const data = uint8arrays.fromString(`${encodedHeader}.${encodedPayload}`)
-  const sig = uint8arrays.fromString(ucan.signature, 'base64url')
+  const header = parseHeader(encodedHeader)
+  const payload = parsePayload(encodedPayload)
 
-  const valid = await verifySignature(data, sig, ucan.payload.iss)
-  if (!valid) return false
-  if (!ucan.payload.prf) return true
+  if (!isUcanHeader(header)) {
+    throw new Error(`Can't parse UCAN header: ${encodedHeader}: Invalid format.`)
+  }
+  if (!isUcanPayload(payload)) {
+    throw new Error(`Can't parse UCAN payload. ${encodedPayload}: Invalid format.`)
+  }
+
+  if (checkSignature) {
+    const data = uint8arrays.fromString(`${encodedHeader}.${encodedPayload}`, "ascii")
```

Comment:
`sign` just does `uint8arrays.fromString(`${encodedHeader}.${encodedPayload}`)` which defaults to `utf8` instead of `ascii`.

all of base64 is both ascii & utf8 so I don't think this should cause any issues, but maybe just standardize on utf8 to be safe?

Reply:
Right. I was going back-and-forth with this a little.
It should definitely be the same in `sign` and `validate`! Good catch.
But I'm still not sure whether it should be `ascii` or `utf8`. Technically non-ascii would have already errored before (since we're both splitting on "." and then parsing the elements using base64url). But of course, that's not really obvious and someone might accidentally change that piece of code/copy it somewhere unsafe. :thinking: I'll take a look at what `ascii` and what `utf8` do. I'd prefer the one that fails less.

Reply:
Okay, neither errors out really. And because `utf8` uses `TextDecoder` underneath which operates on whole strings at a time & is built into the VM, wheras `ascii` iterates through the string and calls `charCodeAt`, it's much slower.
So I'll just remove the `ascii` and let both work with `utf8`!